studies/growable_buffer/growable_buffer_panels.cpp

```diff
@@ -8,54 +8,54 @@
 template <typename PRIMITIVE>
 class GrowableBuffer {
 public:
-  GrowableBuffer(size_t initial) 
-      : initial_(initial) { 
+  GrowableBuffer(size_t initial)
+      : initial_(initial) {
     initial_ = initial;
-    ptr_.push_back(std::make_unique<PRIMITIVE>(initial));
+    ptr_.push_back(std::unique_ptr<PRIMITIVE>(new PRIMITIVE[initial]));
```

Comment:
@ManasviGoyal - please, check if the proposed changes work for you. Thanks!

Reply:
Hi @ianna. The concatenate function still doesn't work but otherwise the changes work for me. Does it work for you?
Also, should I add any other methods from the original GrowableBuffer?

Reply:
> Hi @ianna. The concatenate function still doesn't work but otherwise the changes work for me. Does it work for you?

Yes, please, have a look at the last commit. Also, IMHO we have to move out the index calculations per item. It would be better to make the buffer contiguous first.

> Also, should I add any other methods from the original GrowableBuffer?

I think, you need one more method to update the `ptr_` with a contiguous panel.

Reply:
> Yes, please, have a look at the last commit. Also, IMHO we have to move out the index calculations per item. It would be better to make the buffer contiguous first.

Okay. So will the concatenation be done externally like it is done here in the `main()` or should I just add it inside the `getitem_at_nowrap()`?

> I think, you need one more method to update the `ptr_` with a contiguous panel.

But that can be done by calling `concatenate()` inside the `ptr()` method that we already have just like you suggested earlier. So I don't think we need a separate method for that.